base/src/main/scala/data/ProCompose.scala

```diff
@@ -0,0 +1,54 @@
+package scalaz
+package data 
+
+object ProCompose {
+  import typeclass.{Profunctor, Strong, Category, StrongClass, CategoryClass}
+  import typeclass.Strong._
+  import typeclass.Profunctor._
+
+  trait ProCompose[P[_, _], A, B] { } 
+
+  case class Pure[P[_, _], A, B](pab: A => B) extends ProCompose[P, A, B]
+
+  case class Composed[P[_, _], A, X, B](pab: P[A, X], more: ProCompose[P, X, B]) extends ProCompose[P, A, B]
```

Comment:
I agree, Vector is crap, and so are casts. It's totally possible to make this catenable and type-aligned; but it's not as generic as making the type-aligned sequence (free category over F) and then making this `ProCompose` into a free category over `F :+: Function1`. I feel like free categories are a useful addition to Scalaz 8, but if you disagree I can't see any other issue with making this catenable.